plasma/models/mpi_runner.py

```diff
@@ -792,19 +827,74 @@ def mpi_make_predictions(conf, shot_list, loader, custom_path=None):
             y_prime += y_p
             y_gold += y
             disruptive += disr
-            # print_all('\nFinished with i = {}'.format(i))
 
         if (i % g.num_workers == g.num_workers - 1
                 or i == len(shot_sublists) - 1):
+            # Create numpy block from y list which is used in MPI
+            # Pads y_prime and y_gold with zeros to maximum shot length within block being transferred
+            g.comm.Barrier()
+            if color ==1:
+                shpz = [max(y.shape) for y in y_prime]
+                max_length = max([max(y.shape) for y in y_p])
+                g.write_all(' max length = {}\n'.format(max_length))
+            g.comm.Barrier()
+            max_length = g.comm.allreduce(max_length, MPI.MAX) 
+            if color == 1:
+                y_prime_numpy = np.stack([np.pad(sublist, pad_width=((0,max_length-max(sublist.shape)),(0,0))) for sublist in y_prime])
+                y_gold_numpy = np.stack([np.pad(sublist, pad_width=((0,max_length-max(sublist.shape)),(0,0))) for sublist in y_gold])
+            
+            g.comm.Barrier()
+            temp_predictor_only_comm = MPI.Comm.Split(g.comm, color, i)
             g.comm.Barrier()
```

Comment:
are these MPI Barriers necessary? I would remove all that arent absolutely critical for correctness

Reply:
Probably not. I'll take out the ones which probably don't need to be there and test it

Reply:
Took unessential Barrier's out. Tested with 4 nodes on Traverse. Everything seems ok.

Reply:
@felker, if you're satisfied can you resolve the conversation? If not, what else should happen to test this?